documentation/docs/examples/cad-3dobjects.mdx

```diff
@@ -0,0 +1,120 @@
+viewer.on('click', (event) => {
+  const intersection = viewer.getIntersectionFromPixel(event.offsetX, event.offsetY);
+  if (intersection) {
+    const marker = new THREE.Mesh(markerGeometry, markerMaterial);
+    marker.position.copy(intersection.point);
+    viewer.addObject3D(marker);
+  };
+});
+`}
+</LiveCodeSnippet>
+
+## Add environment
+
+To make a model feel more natural it might be useful to add an environment. The following examples adds sky and a sea effect for the example oil rig. 
+
+First, lets encapsulate the model in a large sky-textured sphere to create an illusion of skies. The sphere map is available from [Free HDR - EXR SKIES-blog](http://freepanorama.blogspot.com/).
+
+<LiveCodeSnippet scope={{skyUrl: useBaseUrl('/img/sky007.jpg')}}>
+{`
+// Skybox
+const skyBox = new THREE.Mesh(
+  new THREE.SphereBufferGeometry(3000, 15, 15), 
+  new THREE.MeshBasicMaterial(
+    {
+      side: THREE.BackSide,
+      // Skybox texture from http://freepanorama.blogspot.com/2010/01/sky7-spherical.html
+      map: new THREE.TextureLoader().load(skyUrl),
+    }
+  )
+);
+viewer.addObject3D(skyBox);
+`}
+</LiveCodeSnippet>
+
+This looks a bit better, but the oil rig appears to be floating mid-air at this point. Lets add an ocean. A plane is rendered at the sea level and an ocean effect added to the plane. The example builds upon [the ocean example from ThreeJS](https://threejs.org/examples/?q=ocea#webgl_shaders_ocean).
+
+Also note that since the ocean has moving waves we want to render the model continuously and not only when camera changes or data is loaded. We do this by repeatedly calling `Cognite3DViewer.forceRerender()`.
+
+<LiveCodeSnippet scope={{Water}}>
+{`
+// The following has been imported outside this scope:
+// import { Water } from 'three/examples/jsm/objects/Water';
+\
+// Water
+var waterGeometry = new THREE.PlaneBufferGeometry(100000, 100000);
+water = new Water(
+  waterGeometry,
+  {
+    textureWidth: 512,
+    textureHeight: 512,
+    waterNormals: new THREE.TextureLoader()
+      .load('https://threejs.org/examples/textures/waternormals.jpg', 
+        texture => {
+          texture.wrapS = texture.wrapT = THREE.RepeatWrapping;
+        }
+      ),
+    waterColor: 0x442e1f,
+  }
+);
+water.rotation.x = - Math.PI / 2;
+viewer.addObject3D(water);
+\
+function updateOcean() {
+  var time = performance.now() * 0.001;
+  water.material.uniforms[ 'time' ].value += 1.0 / 60.0;
+  viewer.forceRerender();
+  requestAnimationFrame(updateOcean);
+};
+updateOcean();
+`}
+</LiveCodeSnippet>
+
+This looks great, but the user is available to move below sea level. Let's restrict the movement to avoid this.
+
+<LiveCodeSnippet>
+{`
+const boundingSphere = new THREE.Sphere(new THREE.Vector3(0,0,0), 2000);
+viewer.on('cameraChange', (position, target) => {
+  // Keep camera above sea level
+  if (position.y < 20) {
+    position.y = 21;
+    viewer.setCameraPosition(position);
+  }
+\
+  // Clamp point to be within bounding sphere
```

Comment:
Well, that might be just me, but I don't understand that part about clamp point. What are the clamp point and the bounding sphere? Some explaining link on 3d basics might be good to have nearby.